src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -98,4 +108,36 @@ public DarCollection updateDarCollection(Integer collectionId, User user) {
     darCollectionDAO.updateDarCollection(collectionId, user.getDacUserId(), new Date());
     return getByCollectionId(collectionId);
   }
+
+  public void addDatasetsToCollections(List<DarCollection> collections) {
+    
+    List<List<DataAccessRequest>> collectionDars = collections.stream()
+      .map(c -> c.getDars())
+      .collect(Collectors.toList());
+    
+    List<Integer> datasetIds = collectionDars.stream()
+      .flatMap(Collection::stream)
+      .map(d -> d.getData().getDatasetIds())
+      .flatMap(Collection::stream)
+      .collect(Collectors.toList());
+
+    Set<DataSet> datasets = datasetDAO.findDatasetWithDataUseByIdList(datasetIds);
+    Map<Integer, DataSet> datasetMap = datasets.stream()
+      .collect(
+        Collectors.toMap(d -> d.getDataSetId(), d -> d)
+      );
+
+    for(int i = 0; i < collections.size(); i++) {
+      DarCollection collection = collections.get(i);
+      List<Integer> collectionDatasetIds = collectionDars.get(i).stream()
+        .map(d -> d.getData().getDatasetIds())
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
+      for(Integer id : collectionDatasetIds) {
+        DataSet dataset = datasetMap.get(id);
+        collection.addDataset(dataset);
+      }
```

Comment:
I'm concerned about the nested loop here, I think we can clean up this logic a bit. Let's pair on this today.